### PR TITLE
Lesson db

### PR DESCRIPTION
**Pull request by NVoronina**

### CODE REVIEW

File: src/main/java/com/example/javaitmo2/repository/CarRepository.java [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
-import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.repository.PagingAndSortingRepository;
 import org.springframework.stereotype.Repository;
 
+import java.util.Collection;
+import java.util.List;
+
 @Repository
-public interface CarRepository extends JpaRepository<CarEntity, Long>  {
```
Comment (lDummyl) [inline src/main/java/com/example/javaitmo2/repository/CarRepository.java:8]: JpaRepo actually extends PagingAndSortingRepo,

File: src/main/java/com/example/javaitmo2/service/CarService.java [hunk 2]
```diff
@@ -23,7 +29,24 @@ public CarService(CarRepository repository, ModelMapper modelMapper) {
     }
 
     public ArrayList<CarResponse> getList() {
```
Comment (lDummyl) [inline src/main/java/com/example/javaitmo2/service/CarService.java]: Its better to use Interface as return type as much as possible, List or Collection in this case

File: src/main/java/com/example/javaitmo2/service/CarService.java [hunk 3]
```diff
@@ -23,7 +29,24 @@ public CarService(CarRepository repository, ModelMapper modelMapper) {
...
+            list = repository.findAllByBrandOrderByIdDesc(brand);
+        } else if (minSeatsCount != null) {
+            Pageable pager = PageRequest.of(page, size, Sort.by("id").descending());
+            list = repository.findAllBySeatsCountGreaterThan(minSeatsCount, pager);
+        } else {
+            throw new ValidationException("No parameters founded");
+        }
         ArrayList<CarResponse> result = new ArrayList<>();
         for (CarEntity car: list) {
             result.add(modelMapper.map(car, CarResponse.class));
```
Comment (lDummyl) [inline src/main/java/com/example/javaitmo2/service/CarService.java]: Try to use functional approach oftenly.
list.stream().map...

File: src/test/java/com/example/javaitmo2/web/CarControllerTests.java [hunk 4]
```diff
@@ -81,7 +84,9 @@ public void shouldReturnedSuccessGetList() throws Exception {
 
     @Test
     public void shouldReturnedSuccessGetOne() throws Exception {
-        CarRequest request = new CarRequest("RRR12TTFGHJS89", "toyota", 5, null);
```
Comment (lDummyl) [inline src/test/java/com/example/javaitmo2/web/CarControllerTests.java:84]: extract as local variable, expectedVin for example. And by the way in asserEquals (expected, testResult), so flip it

File: src/main/java/com/example/javaitmo2/service/CarService.java (hunk 3 above)
Comment (NVoronina) [reply src/main/java/com/example/javaitmo2/service/CarService.java]: +

File: src/test/java/com/example/javaitmo2/web/CarControllerTests.java (hunk 4 above)
Comment (NVoronina) [reply src/test/java/com/example/javaitmo2/web/CarControllerTests.java:84]: +

File: src/main/java/com/example/javaitmo2/service/CarService.java (hunk 2 above)
Comment (NVoronina) [reply src/main/java/com/example/javaitmo2/service/CarService.java]: ++

File: src/main/java/com/example/javaitmo2/repository/CarRepository.java (hunk 1 above)
Comment (NVoronina) [reply src/main/java/com/example/javaitmo2/repository/CarRepository.java:8]: ++

**Pull request merged**